x-pack/elastic-agent/pkg/fleetapi/action.go

```diff
@@ -206,19 +209,24 @@ func (a *ActionApp) Type() string {
 	return a.ActionType
 }
 
+// MarshalMap marshals ActionApp into a corresponding map
+func (a *ActionApp) MarshalMap() (map[string]interface{}, error) {
+	var res map[string]interface{}
+	b, err := json.Marshal(a)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &res)
+	return res, err
```

Comment:
Not a big fan of marshaling/unmarshaling via JSON.

Do we really need a `map[string]interface{}` if we already have a proper type?

Why marshal/unmarshal via JSON? Why don't we just construct the map, or use something like [`mapstructure`](https://github.com/mitchellh/mapstructure), or [`typeconv.Convert`](https://github.com/elastic/beats/blob/master/libbeat/common/transform/typeconv/typeconv.go#L126) (from libbeat, which can work with 'arbitrary' go types)?

Reply:
I understand, I could have coded a map here, but if the underlying structure of the ActionApp changes, would have to remember to update in two places. This way it's only one place.

Reply:
and yes, need map[string]interface{} to pass to already existing 
```
func (as *ApplicationState) PerformAction(name string, params map[string]interface{}, timeout time.Duration) 
```

Reply:
> and yes, need map[string]interface{} to pass to already existing

I see. Still, please consider to use mapstructure or the typeconv package from libbeat.

Reply:
will update